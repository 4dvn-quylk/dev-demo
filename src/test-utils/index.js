import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { createStore } from '../stores/store';

// test utils file
export const renderWithRouter = (ui, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);
  return {
    // user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

export const renderWithProvider = (ui, { route = '/', providerProps } = {}) => {
  window.history.pushState({}, 'Test page', route);
  const AppStore = createStore(providerProps);
  return {
    ...render(<AppStore.Provider>{ui}</AppStore.Provider>, {
      wrapper: BrowserRouter,
    }),
  };
};
