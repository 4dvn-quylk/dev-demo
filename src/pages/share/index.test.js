/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
/* eslint-disable testing-library/prefer-screen-queries */
import '@testing-library/jest-dom';
import { renderWithProvider } from '../../test-utils/index';
import SharePage from './index';

describe('Test Share page', () => {
  test('should render successful', () => {
    const { container } = renderWithProvider(<SharePage />);
    expect(container).toMatchSnapshot();
  });
});
