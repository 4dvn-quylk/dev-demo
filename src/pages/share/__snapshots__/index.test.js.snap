// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Share page should render successful 1`] = `
<div>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-row ant-row-center"
      >
        <div
          class="ant-card ant-card-bordered mt mb"
          style="width: 400px;"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Share a Youtube video
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <form
              autocomplete="off"
              class="ant-form ant-form-horizontal"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-col-8 ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="url"
                    title="Youtube URL"
                  >
                    Youtube URL
                  </label>
                </div>
                <div
                  class="ant-col ant-col-16 ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input"
                        id="url"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="submit"
                      >
                        <span>
                          Submit
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
